checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -575,8 +586,25 @@ private void computeFieldAccessType(AnnotatedTypeMirror type,
                     .asType();
             boolean isInitializedForFrame = isInitializedForFrame(receiverType, fieldDeclarationType);
             if (isInitializedForFrame) {
+                // The receiver is initialized for this frame.
+                // Change the type of the field to @UnknownInitialization so that
+                // anything can be assigned to this field.
                 type.replaceAnnotation(qualHierarchy.getTopAnnotation(UNCLASSIFIED));
+
+            } else if(computingAnnotatedTypeMirrorOfLHS) {
+                // The receiver is not initialized for this frame, and the type of a lhs is being computed.
```

Comment:
Isn't this whole branch equivalent to just doing `type.replaceAnnotation(qualHierarchy.getTopAnnotation(UNCLASSIFIED));`?
The type can only have a single initialization annotation, so the above should replace that one with the top for that hierarchy.